go.mod

```diff
@@ -5,11 +5,12 @@ require (
 	github.com/Azure/azure-sdk-for-go v16.0.0+incompatible
 	github.com/Azure/go-autorest v10.7.0+incompatible
 	github.com/abdullin/seq v0.0.0-20160510034733-d5467c17e7af // indirect
-	github.com/aws/aws-sdk-go v1.15.24
+	github.com/aws/aws-sdk-go v1.15.41
 	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/denverdino/aliyungo v0.0.0-20170926055100-d3308649c661
 	github.com/dgrijalva/jwt-go v3.2.0+incompatible // indirect
 	github.com/digitalocean/godo v1.1.1
+	github.com/dimchansky/utfbom v1.1.0 // indirect
```

Comment:
This is unrelated change but it looks like it's a dependency of `github.com/Azure/go-autorest` v.10.7.0 [1] but somehow we missed declaring it here.

[1] https://github.com/Azure/go-autorest/blob/v10.7.0/autorest/azure/auth/auth.go#L34

Reply:
Is this the most conservative update that includes the things we need?

FWIW Consul now pulls in `1.25.42` anyway thanks to recent AWS CA provider work and we tested go-discover integration afterwards and it works.

Reply:
Oops - I had updated it to `v1.25.41` the most recent release with `ec2metadata` changes[1].  I accidentally updated it with `v1.15.41` earlier.

[1] https://github.com/aws/aws-sdk-go/releases/

Reply:
Correction: Consul pulls in `v1.25.32` which is what I tested with go-discover stuff.

Let me sanity check Consul AWS cloud auto join with 1.25.41 and then we can call this good.

Reply:
Yep works with 1.25.41:

```bash
$ go get github.com/aws/aws-sdk-go@v1.25.41
...
$ make dev
$ envchain csl consul agent -retry-join "provider=aws tag_key=Role tag_value=consul-server region=us-west-1" -data-dir "/tmp/consul"
...
2019/12/02 15:26:47 [INFO] agent: Discovered LAN servers: 10.0.1.241 10.0.1.174 10.0.1.12 10.0.1.33 10.0.1.237 10.0.1.44 10.0.1.11 10.0.1.156 10.0.1.28
...
```

(We have dev servers running in that region so this discovered those and could join them.